tests/app/views/contexts/test_calculated_summary_context.py

```diff
@@ -9,237 +7,115 @@
-        )
-
-        self.assertTrue("calculated_question" in context_summary)
-        self.assertEqual(
-            context_summary["calculated_question"]["title"],
-            "Grand total of previous values",
-        )
-        self.assertEqual(
-            context_summary["calculated_question"]["answers"][0]["value"], "20%"
-        )
-
-    @patch("app.jinja_filters.flask_babel.get_locale", Mock(return_value="cy"))
-    def test_build_view_context_for_number_calculated_summary(self):
-        current_location = Location(
-            section_id="default-section", block_id="number-total-playback"
-        )
-
-        calculated_summary_context = CalculatedSummaryContext(
+            "20%",
+            2,
+            [],
+        ),
+        (
+            "number-total-playback",
+            "cy",
             "cy",
-            self.schema,
-            self.answer_store,
-            self.list_store,
-            self.progress_store,
-            self.metadata,
-            self.response_metadata,
-        )
-
-        context = calculated_summary_context.build_view_context_for_calculated_summary(
-            current_location
-        )
-
-        self.assertIn("summary", context)
-        self.assert_summary_context(context)
-        self.assertEqual(len(context["summary"]), 6)
-        context_summary = context["summary"]
-        self.assertTrue("title" in context_summary)
-        self.assertEqual(
-            context_summary["title"],
             "We calculate the total of number values entered to be 22. Is this correct?",
-        )
+            "22",
+            2,
+            [],
+        ),
+    ),
+)
+def test_build_view_context_for_currency_calculated_summary_no_skip(
+    block_id,
+    locale,
+    language,
+    title,
+    value,
+    block_len,
+    answers,
+    summary_context_schema,
+    summary_context_answers,
+    list_store,
+    progress_store,
+    mocker,
+):
+    mocker.patch(
+        "app.jinja_filters.flask_babel.get_locale",
+        mocker.MagicMock(return_value=locale),
+    )
+
+    current_location = Location(section_id="default-section", block_id=block_id)
+
+    calculated_summary_context = CalculatedSummaryContext(
+        language,
+        summary_context_schema,
+        summary_context_answers,
+        list_store,
+        progress_store,
+        {},
+        {},
+    )
 
-        self.assertTrue("calculated_question" in context_summary)
-        self.assertEqual(
-            context_summary["calculated_question"]["title"],
-            "Grand total of previous values",
-        )
-        self.assertEqual(
-            context_summary["calculated_question"]["answers"][0]["value"], "22"
-        )
+    context = calculated_summary_context.build_view_context_for_calculated_summary(
+        current_location
+    )
+
+    for answer in answers:
+        summary_context_answers.add_or_update(answer)
+
+    assert "summary" in context, "Key value summary missing from context"
```

Comment:
Confused as to why, some asserts have the exception text defined and some doesn't?. Don't think they really add much in these simple scenarios.